tests/Doctrine/Tests/DBAL/Types/JsonTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Types;
+
+use Doctrine\DBAL\Types\Type;
+use Doctrine\Tests\DBAL\Mocks\MockPlatform;
+
+class JsonTest extends \Doctrine\Tests\DbalTestCase
+{
+    /**
+     * @var \Doctrine\Tests\DBAL\Mocks\MockPlatform
+     */
+    protected $platform;
+
+    /**
+     * @var \Doctrine\DBAL\Types\JsonType
+     */
+    protected $type;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp()
+    {
+        $this->platform = new MockPlatform();
+        $this->type     = Type::getType('json');
+    }
+
+    public function testReturnsBindingType()
+    {
+        $this->assertSame(\PDO::PARAM_STR, $this->type->getBindingType());
+    }
+
+    public function testReturnsName()
+    {
+        $this->assertSame(Type::JSON_UNFLAWED, $this->type->getName());
+    }
+
+    public function testReturnsSQLDeclaration()
+    {
+        $this->assertSame('DUMMYJSON', $this->type->getSQLDeclaration(array(), $this->platform));
+    }
+
+    public function testJsonNullConvertsToPHPValue()
+    {
+        $this->assertNull($this->type->convertToPHPValue(null, $this->platform));
+    }
+
+    public function testJsonEmptyStringConvertsToPHPValue()
+    {
+        $this->assertNull($this->type->convertToPHPValue('', $this->platform));
+    }
+
+    public function testJsonStringConvertsToPHPValue()
+    {
+        $value         = array('foo' => 'bar', 'bar' => 'foo');
+        $databaseValue = json_encode($value);
+        $phpValue      = $this->type->convertToPHPValue($databaseValue, $this->platform);
+
+        $this->assertEquals($value, $phpValue);
+    }
+
+    /** @dataProvider providerFailure */
+    public function testConversionFailure($data)
+    {
+        error_reporting( (E_ALL | E_STRICT) - \E_NOTICE );
```

Comment:
Well, I would remove it then, to be sure we don't get a notice by mistake and uncaught by the testsuite